tests/TestCase.php

```diff
@@ -59,6 +59,27 @@ protected function expectCallableConsecutive($numberOfCalls, array $with)
         return $mock;
     }
 
+    protected function expectCallableConsecutiveWillReturnCallback(array $with, $returnCallback)
+    {
+        $mock = $this->createCallableMock();
+
+        // Allows to pass either a single callback or an array of callbacks
+        // where the indexes match the indexes of $with
+        if (!is_array($returnCallback)) {
+            $returnCallback = array_fill(0, count($with), $returnCallback);
+        }
+
+        for ($i = 0, $l = count($with); $i < $l; $i++) {
+            $mock
+                ->expects($this->at($i))
+                ->method('__invoke')
+                ->with($this->equalTo($with[$i]))
+                ->willReturnCallback($returnCallback[$i]);
+        }
+
+        return $mock;
+    }
```

Comment:
How would we test this method? :yum: 

On a more serious note, this looks really complicated for such a simple test. As an alternative, maybe keep a counter somewhere and explicitly call the handler twice?

Reply:
This is a copy of the `expectCallableConsecutive()` from above with additional `$mock->willReturnCallback()` ;)

> As an alternative, maybe keep a counter somewhere and explicitly call the handler twice?

Not sure what you mean here? Could you elaborate?

Reply:
@clue I've updated the test and remove the test helper.